.github/workflows/docker.yml

```diff
@@ -13,106 +15,136 @@ env:
+        needs: [build-docker-single-arch, extract-version]
+        strategy:
+            matrix:
+                modernity: ["", "-modern"]
         env:
+            # We need to enable experimental docker features in order to use `docker manifest`
             DOCKER_CLI_EXPERIMENTAL: enabled
+            VERSION: ${{ needs.extract-version.outputs.VERSION }}
+            VERSION_SUFFIX: ${{ needs.extract-version.outputs.VERSION_SUFFIX }}
         steps:
             - name: Dockerhub login
               run: |
                   echo "${DOCKER_PASSWORD}" | docker login --username ${DOCKER_USERNAME} --password-stdin
-            - name: Set Env
-              if: needs.extract-branch-name.outputs.BRANCH_NAME == 'unstable'
-              run: |
-                  echo "TAG_SUFFIX=-unstable" >> $GITHUB_ENV;
             - name: Create and push multiarch manifest
               run: |
-                  docker manifest create ${IMAGE_NAME}:latest${TAG_SUFFIX} \
-                      --amend ${IMAGE_NAME}:latest-arm64${TAG_SUFFIX} \
-                      --amend ${IMAGE_NAME}:latest-amd64${TAG_SUFFIX};
-                  docker manifest push ${IMAGE_NAME}:latest${TAG_SUFFIX}
+                  docker manifest create ${IMAGE_NAME}:${VERSION}${VERSION_SUFFIX}${{ matrix.modernity }} \
+                      --amend ${IMAGE_NAME}:${VERSION}-arm64${VERSION_SUFFIX}${{ matrix.modernity }} \
+                      --amend ${IMAGE_NAME}:${VERSION}-amd64${VERSION_SUFFIX}${{ matrix.modernity }};
+                  docker manifest push ${IMAGE_NAME}:${VERSION}${VERSION_SUFFIX}${{ matrix.modernity }}
     build-docker-lcli:
         runs-on: ubuntu-18.04
-        needs: [extract-branch-name]
+        needs: [extract-version]
+        env:
+            VERSION: ${{ needs.extract-version.outputs.VERSION }}
+            VERSION_SUFFIX: ${{ needs.extract-version.outputs.VERSION_SUFFIX }}
         steps:
             - uses: actions/checkout@v2
             - name: Dockerhub login
               run: |
                   echo "${DOCKER_PASSWORD}" | docker login --username ${DOCKER_USERNAME} --password-stdin
-            - name: Set Env
-              if: needs.extract-branch-name.outputs.BRANCH_NAME == 'unstable'
-              run: |
-                  echo "TAG_SUFFIX=-unstable" >> $GITHUB_ENV;
             - name: Build lcli dockerfile (with push)
               run: |
                   docker build \
                       --build-arg PORTABLE=true \
-                      --tag ${LCLI_IMAGE_NAME}:latest${TAG_SUFFIX} \
+                      --tag ${LCLI_IMAGE_NAME}:${VERSION}${VERSION_SUFFIX} \
                       --file ./lcli/Dockerfile .
-                  docker push ${LCLI_IMAGE_NAME}:latest${TAG_SUFFIX}
+                  docker push ${LCLI_IMAGE_NAME}:${VERSION}${VERSION_SUFFIX}
+    # This job just tests the `Dockerfile` without publishing any images.
+    build-docker-from-source:
```

Comment:
Ah, good catch!

I'm tempted to keep it in `test-suite` because then it will run on pull requests rather than just `unstable` & release. I've consolidated them in `test-suite` for now

Reply:
Good point! agree on keeping it in `test-suite` then